core/src/main/java/com/amplifyframework/core/model/AuthProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * The type of provider for an @auth directive rule.
+ * @see <a href="https://docs.amplify.aws/cli/graphql-transformer/directives#auth">
+ *     GraphQL Transformer @auth directive documentation.</a>
+ */
+public enum AuthProvider {
```

Comment:
Currently this is being proposed for the Core module, at `com.amplifyframework.core.model`. That doesn't seem right. How can we get this out of Core, and into a more purpose-built package?